aws.go

```diff
@@ -137,6 +143,28 @@ func (l *awsLogger) Errorf(ctx context.Context, format string, v ...any) {
 	l.log(ctx, slog.LevelError, fmt.Sprintf(format, v...))
 }
 
+// AddAttributes adds attributes to include in middleware-driven logs
+func (l *awsLogger) AddAttributes(attrbs map[string]any) {
+	l.mu.Lock()
+	if l.attributes == nil {
+		l.attributes = make(map[string]any)
+	}
```

Comment:
We should do this initialization in `newAWSLogger()` instead of checking this each time.